### ISSUE TITLE
Encrypted files not decrypted

### ISSUE THREAD

**Issue by sy-gpe** [nextcloud/groupfolders#2793]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Upload a file to a group folder with encryption enabled
2. Download the file back

### Expected behaviour
The file should be decrypted before download.

### Actual behaviour
The client is downloading the encrypted file.

### Server configuration

**Operating system:** Rocky Linux 9

**Web server:** HTTPD

**Database:** MariaDB

**PHP version:** 8.2

**Nextcloud version:** (see Nextcloud admin page) 28.0.1

**Group folders version:** 16.0.2 (16.0.3 also affected)

**Updated from an older Nextcloud/ownCloud or fresh install:** Tried on both an updated instance and a newly deployed POC.

**Where did you install Nextcloud from:** Official nextcloud repo

**Are you using external storage, if yes which one:** No

**Are you using encryption:** yes

**Are you using an external user-backend, if yes which one:** SAML on prod instance. Local users on POC.

### Client configuration
**Browser:** Mozilla Firefox

**Operating system:** Linux

#### Web server error log
<details>
<summary>Web server error log</summary>

```
[Thu Feb 01 15:06:58.487125 2024] [core:notice] [pid 9279:tid 9279] SELinux policy enabled; httpd running as context system_u:system_r:httpd_t:s0
[Thu Feb 01 15:06:58.488475 2024] [suexec:notice] [pid 9279:tid 9279] AH01232: suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
AH00558: httpd: Could not reliably determine the server's fully qualified domain name, using localhost.localdomain. Set the 'ServerName' directive globally to suppress this message
[Thu Feb 01 15:06:58.499289 2024] [lbmethod_heartbeat:notice] [pid 9279:tid 9279] AH02282: No slotmem from mod_heartmonitor
[Thu Feb 01 15:06:58.502286 2024] [mpm_event:notice] [pid 9279:tid 9279] AH00489: Apache/2.4.57 (Rocky Linux) configured -- resuming normal operations
[Thu Feb 01 15:06:58.502306 2024] [core:notice] [pid 9279:tid 9279] AH00094: Command line: '/usr/sbin/httpd -D FOREGROUND'
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
No logs generated.
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Not relevant, server side issue.
```
</details>

### More informations

Hello !

I have found out that the changes to the `getScanner` function in PR #2730 has introduced this issue as reverting the function to what it was before fixes the problem.

Could you have a look on this issue ?

I will stay available if you need more informations.

Thanks in advance !

**Comment by landshuterloewe** [comment]
Same issue here.

**Comment by icewind1991** [comment]
should be fixed with 16.0.4
